runcoms/zshenv

```diff
@@ -6,6 +6,6 @@
 #
 
 # Ensure that a non-login, non-interactive shell has a defined environment.
-if [[ "$SHLVL" -eq 1 && ! -o LOGIN && -s "${ZDOTDIR:-$HOME}/.zprofile" ]]; then
+if [[ "$SHLVL" -eq 1 && ! -o LOGIN && -s "${ZDOTDIR:-$HOME}/.zprofile" || ! "${TMPDIR}" ]]; then
```

Comment:
It will only be sourced if that file exists AND `[[ "$SHLVL" -eq 1 && ! -o LOGIN ]]`

Looking at it again though, this could cause it to try and source a non-existant file. I will make it so it requires .zprofile even when TMPDIR isn't set.